frontend/src/metabase/core/components/EditableText/EditableText.unit.spec.tsx

```diff
@@ -0,0 +1,83 @@
+import React from "react";
+import userEvent from "@testing-library/user-event";
+
+import { screen, render, waitFor } from "__support__/ui";
+import EditableText, { EditableTextProps } from "./EditableText";
+
+const setup = (props?: Partial<EditableTextProps>) => {
+  render(<EditableText {...props} />);
+};
+
+describe("EditableText", () => {
+  describe("when there is no initial value", () => {
+    it("should render input", () => {
+      setup();
+
+      expect(screen.getByRole("textbox")).toBeInTheDocument();
+    });
+  });
+
+  it("should render markdown, not input", () => {
+    setup({
+      initialValue: "Description",
+      isMarkdown: true,
+    });
+
+    expect(screen.queryByRole("textbox")).not.toBeInTheDocument();
+    expect(screen.getByText("Description")).toBeInTheDocument();
+  });
+
+  it("should render input after a click", () => {
+    setup({
+      initialValue: "Description",
+      isMarkdown: true,
+    });
+
+    userEvent.click(screen.getByText("Description"));
+
+    expect(screen.getByRole("textbox")).toBeInTheDocument();
+  });
+
+  it("should focus the input", async () => {
+    setup({
+      initialValue: "Description",
+      isMarkdown: true,
+    });
+
+    userEvent.click(screen.getByText("Description"));
+
+    await waitFor(() => {
+      expect(screen.getByRole("textbox")).toHaveFocus();
+    });
+  });
+
+  it("should render markdown on blur", async () => {
+    setup({
+      initialValue: "**bold** [link](https://metabase.com)",
+      isMarkdown: true,
+    });
+
+    userEvent.click(screen.getByTestId("editable-text"));
```

Comment:
Good point. I checked aria-label and seems it makes sense to put it only on interactive elements like input, it's not possible in this case. at the same time I checked `aria-expanded` which seems more appropriate for our case, but I also find it confusing.

Reply:
> It's worth noting that ARIA attributes are primarily intended for accessibility purposes, and using them for testing purposes can make the tests less accessible

Reply:
> > It's worth noting that ARIA attributes are primarily intended for accessibility purposes, and using them for testing purposes can make the tests less accessible

Mind sharing where that excerpt is from? I'm also learning about accessibility.

I found [examples from this w3c wiki](https://www.w3.org/WAI/GL/wiki/Using_aria-label_to_provide_labels_for_objects) which sets `aria-label` on `div`s also [from MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label)

> Note: `aria-label` is intended for use on interactive elements, or elements made to be interactive via other ARIA declarations, when there is no appropriate text visible in the DOM that could be referenced as a label

Maybe that's in line with the examples from w3c that have `role` on `div`s?

Anyway, I know that our div isn't interactive per se, but it's meant to be interactive IMO since we want to be able to click on it.